go-controller/observability-lib/sampledecoder/sample_decoder.go

```diff
@@ -0,0 +1,290 @@
+package sampledecoder
+
+import (
+	"bytes"
+	"context"
+	"encoding/binary"
+	"fmt"
+
+	"github.com/ovn-org/libovsdb/client"
+	"github.com/ovn-org/ovn-kubernetes/go-controller/observability-lib/ovsdb"
+	libovsdbops "github.com/ovn-org/ovn-kubernetes/go-controller/pkg/libovsdb/ops"
+	"github.com/ovn-org/ovn-kubernetes/go-controller/pkg/nbdb"
+	"github.com/ovn-org/ovn-kubernetes/go-controller/pkg/observability"
+)
+
+type SampleDecoder struct {
+	nbClient          client.Client
+	ovsdbClient       client.Client
+	cleanupCollectors []int
+}
+
+type nbConfig struct {
+	address string
+	scheme  string
+}
+
+type Cookie struct {
+	ObsDomainID uint32
+	ObsPointID  uint32
+}
+
+const CookieSize = 8
+const bridgeName = "br-int"
+
+var SampleEndian = getEndian()
+
+func getEndian() binary.ByteOrder {
+	// Use network bite order
+	return binary.BigEndian
+}
+
+func getLocalNBClient(ctx context.Context) (client.Client, error) {
+	config := nbConfig{
+		address: "unix:/var/run/ovn/ovnnb_db.sock",
+		scheme:  "unix",
+	}
```

Comment:
This restricts the support to IC (single node per zone) deployments.  I understand it's more work to make this more generic but it should be possible to allow connections to remote NBs.  The rest of the ovnkube components can connect to remote SBs.  Would it make sense to add optional command line arguments to ovnkube-observ to allow the users to pass a custom NB connection string (and potentially more SSL related arguments) in order to support non-IC too?

Reply:
I think we should at least document this limitation in `ovn-observability.md`.

Reply:
that is a good point, I plan to add remote nbdb option in the future, when we have e2es for this feature to make sure that option works. As of now only local nbdb option will be used and tested downstream, I will add this note to the docs.

Reply:
Ack, thanks.